app/code/Magento/Store/Model/StoreRepository.php

```diff
@@ -104,13 +106,15 @@ public function getById($id)
         }
 
         $storeData = $this->getAppConfig()->get('scopes', "stores/$id", []);
-        $store = $this->storeFactory->create([
-            'data' => $storeData
-        ]);
+        $store = $this->storeFactory->create(
+            [
+                'data' => $storeData
+            ]
+        );
 
         if ($store->getId() === null) {
             throw new NoSuchEntityException(
-                __("The store that was requested wasn't found. Verify the store and try again.")
+                __("The store with id $id that was requested wasn't found. Verify the store and try again.")
```

Comment:
@avoelkl Does this error message appear in the browser (admin and/or website)? If yes, can you please escape the output of `$id`.

Reply:
Hi, I've added a cast to (int) to the id.